statsmodels/graphics/functional.py

```diff
@@ -1,14 +1,315 @@
+    alpha.sort(reverse=True)
+
+    n_percentiles = len(alpha)
+    pdf_r = ks_gaussian.pdf(data_r).flatten()
+    pvalues = [np.percentile(pdf_r, (1 - alpha[i]) * 100,
+                             interpolation='linear')
+               for i in range(n_percentiles)]
+
+    # Find mean, quartiles and outliers curves
+    median = pdf.argmax()
+    median = contour_stack[median]
+
+    outliers = np.where(pdf_r < pvalues[alpha.index(threshold)])
+    labels = labels[outliers]
+    outliers = data[outliers]
+
+    extreme_percentile = np.where((pdf > pvalues[alpha.index(0.9)])
+                                  & (pdf < pvalues[alpha.index(0.5)]))
+    extreme_percentile = contour_stack[extreme_percentile]
+
+    mean_percentile = np.where(pdf > pvalues[alpha.index(0.5)])
+    mean_percentile = contour_stack[mean_percentile]
+
+    extra_alpha = [i for i in alpha
+                   if 0.5 != i and 0.9 != i and threshold != i]
+    if extra_alpha != []:
+        extra_percentiles = []
+        for i in extra_alpha:
+            extra_percentile = np.where(pdf > pvalues[alpha.index(i)])
+            extra_percentile = contour_stack[extra_percentile]
+            extra_percentile = inverse_transform_(pca, extra_percentile)
+            extra_percentiles.extend([extra_percentile.max(axis=0),
+                                      extra_percentile.min(axis=0)])
+    else:
+        extra_percentiles = None
+
+    # Inverse transform from bivariate plot to dataset
+    median = inverse_transform_(pca, median)[0]
+    extreme_percentile = inverse_transform_(pca, extreme_percentile)
+    mean_percentile = inverse_transform_(pca, mean_percentile)
+
+    extreme_percentile = [extreme_percentile.max(axis=0),
+                          extreme_percentile.min(axis=0)]
+    mean_percentile = [mean_percentile.max(axis=0),
+                       mean_percentile.min(axis=0)]
+
+    hdr_res = {
+        "median": median,
+        "mean_percentile": mean_percentile,
+        "extreme_percentile": extreme_percentile,
+        "extra_percentiles": extra_percentiles,
+        "outliers": outliers
+    }
+
+    # Plots
+    ax.plot(np.array([xdata] * n_samples).T, data.T,
+            c='c', alpha=.1, label='dataset')
+    ax.fill_between(xdata, *mean_percentile,
+                    color='gray', alpha=.4,  label='50th percentile')
+    ax.fill_between(xdata, *extreme_percentile,
+                    color='gray', alpha=.3, label='90th percentile')
+
+    try:
+        ax.plot(np.array([xdata] * len(extra_percentiles)).T,
+                np.array(extra_percentiles).T,
+                c='y', ls='-.', alpha=.4, label='Extra percentiles')
+    except TypeError:
+        pass
+
+    ax.plot(xdata, median, c='k')
+
+    try:
+        for ii, outlier in enumerate(outliers):
+            label = str(labels[ii]) if labels is not None else None
+            ax.plot(xdata, outlier,
+                    ls='--', alpha=0.7, label=label)
+    except ValueError:
```

Comment:
what raises the ValueError?
An explicit check would make the code more informative and more robust.
e.g. `if len(outliers) != 0`